packages/perseus/src/widgets/interactive-graphs/mafs-graph.tsx

```diff
@@ -396,73 +396,120 @@ const renderPointGraphControls = (props: {
-                    tabIndex={-1}
-                    style={{
-                        width: "100%",
-                        marginLeft: "20px",
-                        visibility: shouldShowRemoveButton
-                            ? "visible"
-                            : "hidden",
-                    }}
-                    onClick={(_event) => {
-                        props.dispatch(
-                            actions.polygon.removePoint(
-                                props.state.focusedPointIndex!,
-                            ),
-                        );
-                    }}
-                >
-                    {perseusStrings.removePoint}
-                </Button>
-            )}
-        </View>
+            {perseusStrings.openPolygon}
+        </Button>
+    ) : (
+        <Button
+            kind="secondary"
+            // Conditional disable when there are less than 3 points in
+            // the graph
+            disabled={coords.length < 3}
+            style={{
+                width: "100%",
+                marginLeft: "20px",
+            }}
+            tabIndex={0}
+            onClick={() => {
+                props.dispatch(actions.polygon.closePolygon());
+            }}
+        >
+            {perseusStrings.closePolygon}
+        </Button>
+    );
+
+    return (
+        <>
+            <View
+                style={{
+                    flexDirection: "row",
+                    width: props.width,
+                }}
+            >
+                {/**
+                 * Only show this in keyboard mode.
+                 */}
+                {interactionMode === "keyboard" && (
+                    <Button
+                        kind="secondary"
+                        style={{
+                            width: "100%",
+                            marginLeft: "20px",
+                        }}
+                        // Disable button when polygon is closed.
+                        disabled={closedPolygon}
+                        // Do not make the button tabbable when it is disabled.
+                        tabIndex={closedPolygon ? -1 : 0}
+                        onClick={() => {
+                            props.dispatch(actions.polygon.addPoint([0, 0]));
+                        }}
+                    >
+                        {perseusStrings.addPoint}
+                    </Button>
+                )}
+                {/*
+                    Make sure remove button is always present, just disabled/enabled depending
+                    on when a point is selected or if the polygon is closed.
+                */}
+                {interactionMode === "mouse" && (
+                    <Button
+                        id={REMOVE_BUTTON_ID}
+                        kind="secondary"
+                        color="destructive"
+                        // Disable button when polygon is closed.
+                        disabled={closedPolygon || !shouldShowRemoveButton}
```

Comment:
Re: my comment above, maybe `closedPolygon` should be incorporated into `shouldShowRemoveButton`/`shouldEnableRemoveButton` so all the logic can be in one place.